lib/hsa/hc_am.cpp

```diff
@@ -524,8 +528,9 @@ am_status_t am_map_to_peers(void* ptr, size_t num_peer, const hc::accelerator* p
         else
             return AM_ERROR_MISC;
     }
-    
-    std::vector<hsa_agent_t> agents{hc::accelerator::get_all().size()};
+
+    const std::size_t max_agent = hc::accelerator::get_all().size();
+    hsa_agent_t agents[max_agent];
```

Comment:
please resync with `clang_tot_upgrade` branch. there is a recent PR which fixed this VLA.